### ISSUE TITLE
heterogeneous softmax

### ISSUE THREAD

**Issue by Tomposon** [THUDM/GATNE#12]
请问一下目前的代码是不是在训练的时候没有考虑node type

1.inductive setting里的节点的feature转换矩阵h在论文中带有下标z ， 是不是意味着h应该和node type相关，但是代码实现好像对于所有的节点都使用同一个转换矩阵。
2.目前的softmax是不是还是普通的softmax，没有考虑node type，代码中tf.nn.nce_loss函数里num_class对应的是所有的节点数目，对于heterogeneous softmax是不是应该要根据不同的node type，给出不同的num_class值啊。

感谢

**Comment by cenyk1230** [comment]
1. 我们在生成feature的时候做了处理（统一了feature的维数，用0来填充不属于自己的node type的feature）。
2. 现在的代码里用的是普通的softmax。heterogeneous softmax是从metapath2vec中直接借用过来的。从metapath2vec的文章中report的结果来看，其实效果是差不多的。

**Comment by Tomposon** [comment]
好的，非常感谢你的解答！
